### PR TITLE
Fix rethinkdb plugin to work with actual database and actual authorization protocol

### PR DESCRIPTION
**Pull request by vodolaz095**
### Required for all PRs:

- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by vodolaz095** [comment]
Previous PR - #2959

**Comment by vodolaz095** [comment]
Issue related - https://github.com/influxdata/telegraf/issues/2958

**Comment by vodolaz095** [comment]
code seems working - it passed your CI tests, and it works ok on my local machine

**Comment by danielnelson** [comment]
What do you think about my suggestion to add username and password options to the plugin, and then use the old handshake method if authkey is set in the url, otherwise use the username and password?  

I'd like to move away from the url encoded userinfo since it can be tricky to escape special characters, makes logging difficult, and its use is deprecated in rfc3986.

**Comment by vodolaz095** [comment]
i think DSL names is best approach, universal - it is possible to monitor servers with different versions and different credentials easily:

```
[[inputs.rethinkdb]]
  servers = [
# server without authorization, like old behaviour
    "rethinkdb.local:28015",
# servers WITH old authkey authorization, old behaviour, 
# using Handshake v0.4 protocol for authorization
    "username:auth_key@rethinkdb.local:28015",
    "rethinkdb://username:authkey@rethinkdb.local:28015",

# servers with actual Handshake v1 protocol - 
# username and password authorization
# using - new behaviour
    "rethinkdb2://username:authkey@rethinkdb2.local:28015",
]
```

so this approach allows to monitor both old and new versions on rethinkdb, and is backward compatible with approach being used before.

anyway, for me i plan to use approach i made in PR or in my fork.

maybe servers have to be an array of hashes consisting of username, password, hostname, port, authkey, protocol to use

**Comment by danielnelson** [comment]
Oh yeah, my way is problematic with multiple servers and we don't want to make breaking changes to the config for this, so lets just go with this.  Perhaps in the future we can have one server per plugin instance.

File: plugins/inputs/rethinkdb/rethinkdb.go [hunk 1]
```diff
@@ -8,20 +8,30 @@ import (
...
+#   ##   10.0.0.1:10000, etc.
+#   servers = ["127.0.0.1:28015"]
+#   ##
+#   ## If you use actual rethinkdb of > 2.3.0 with username/password authorization,
+#   ## protocol have to be named "rethinkdb2" - it will use 1_0 H.
+#   servers = ["rethinkdb2://username:password@127.0.0.1:28015"]
+#   ##
+#   ## If you use older versions of rethinkdb (<2.2) with auth_key, protocol
+#   ## have to be named "rethinkdb".
+#   servers = ["rethinkdb://username:auth_key@127.0.0.1:28015"]
```
Comment (danielnelson) [inline plugins/inputs/rethinkdb/rethinkdb.go]: For the sampleConfig, don't have the header and remove the top level comment #.  Test out how it looks with `telegraf --input-filter rethinkdb --output-filter none config`

**Comment by vodolaz095** [comment]
thx!

**Comment by danielnelson** [comment]
Thanks so much!

**Pull request merged**